File: 01-week-1--starter-algorithms/02-day-3--fibonacci-series/javascript/fibonacci.js
function fibonacci(num) {
  if (num < 2) {
    return num
  }
  let twoNum = [0, 1]
  for (let i = 0; i < num-1; ++i) {
    const sum = twoNum[0] + twoNum[1]
    twoNum = [twoNum[1], sum]
  }
  return twoNum[1]
}

if (require.main === module) {
  // add your own tests in here
  console.log("Expecting: 0");
  console.log("=>", fibonacci(0));

  console.log("");

  console.log("Expecting: 1");
  console.log("=>", fibonacci(2));

  console.log("");

  console.log("Expecting: 55");
  console.log("=>", fibonacci(10));

  console.log("");

  console.log("Expecting: 5");
  console.log("=>", fibonacci(4));
}

module.exports = fibonacci;

// Please add your pseudocode to this file
// And a written explanation of your solution

File: 01-week-1--starter-algorithms/02-day-3--fibonacci-series/ruby/fibonacci.rb
def fibonacci(num)
  two_num = [0, 1]
  return two_num[num] if num < 2
  (num - 1).times do
    sum = two_num[0] + two_num[1]
    two_num = [two_num[1], sum]
  end
  two_num.last
end

if __FILE__ == $PROGRAM_NAME
  puts "Expecting: 0"
  puts "=>", fibonacci(0)

  puts

  puts "Expecting: 1"
  puts "=>", fibonacci(2)

  puts

  puts "Expecting: 55"
  puts "=>", fibonacci(10)

  puts

  puts "Expecting: 3"
  puts "=>", fibonacci(4)

  # Don't forget to add your own!
end

# Please add your pseudocode to this file
# array with two values then iterate through the sums
# And a written explanation of your solution
